Source/Renderer/modernizeShader.js

```diff
@@ -0,0 +1,227 @@
+define([
+        '../Core/defined',
+        '../Core/DeveloperError',
+    ], function(
+        defined,
+        DeveloperError) {
+    'use strict';
+
+    /**
+     * A function to port GLSL shaders from GLSL ES 1.00 to GLSL ES 3.00
+     *
+     * This function is nowhere near comprehensive or complete. It just
+     * handles some common cases.
+     *
+     * Note that this function requires the presence of the
+     * "#define OUTPUT_DECLARATION" line that is appended
+     * by ShaderSource.
+     *
+     * @private
+     */
+    function modernizeShader(source, isFragmentShader) {
+        var outputDeclarationRegex = /#define OUTPUT_DECLARATION/;
+        var splitSource = source.split('\n');
+
+        if (/#version 300 es/g.test(source)) {
+            return source;
+        }
+
+        var outputDeclarationLine = -1;
+        var i, line;
+        for (i = 0; i < splitSource.length; ++i) {
+            line = splitSource[i];
+            if (outputDeclarationRegex.test(line)) {
+                outputDeclarationLine = i;
+                break;
+            }
+        }
+
+        if (outputDeclarationLine === -1) {
+            throw new DeveloperError('Could not find a #define OUTPUT_DECLARATION!');
```

Comment:
Just end with a period; I don't think any of our other exceptions alarm users with a bang (`!`). :smile: